src/components/NewAnalysisModal.js

```diff
@@ -1,37 +1,106 @@
+import { div, h, h2, hr, img, span } from 'react-hyperscript-helpers'
+import { ButtonPrimary, IdContainer, Select, spinnerOverlay, WarningTitle } from 'src/components/common'
+import Dropzone from 'src/components/Dropzone'
+import { icon } from 'src/components/icons'
 import ModalDrawer from 'src/components/ModalDrawer'
 import { NewGalaxyModalBase } from 'src/components/NewGalaxyModal'
 import { NewRuntimeModalBase } from 'src/components/NewRuntimeModal'
-import { tools } from 'src/components/notebook-utils'
+import { analysisNameInput, analysisNameValidator, getDisplayName, getTool, notebookData, tools } from 'src/components/notebook-utils'
 import TitleBar from 'src/components/TitleBar'
 import galaxyLogo from 'src/images/galaxy-logo.png'
 import jupyterLogoLong from 'src/images/jupyter-logo-long.png'
 import rstudioLogo from 'src/images/rstudio-logo.svg'
+import { Ajax } from 'src/libs/ajax'
 import colors from 'src/libs/colors'
-import { withErrorReporting } from 'src/libs/error'
+import { reportError, withErrorReporting } from 'src/libs/error'
+import { FormLabel } from 'src/libs/forms'
+import { getCurrentApp, getCurrentRuntime, isRuntimeDeletable } from 'src/libs/runtime-utils'
+import * as Style from 'src/libs/style'
 import * as Utils from 'src/libs/utils'
+import validate from 'validate.js'
 
-//TODO: title id in titleBar and aria-labelled-by in withModalDrawer
 
 const titleId = 'new-analysis-modal-title'
+const NEW_ANALYSIS_MODE = 'NEW_ARTIFACT'
+const NEW_ENVIRONMENT_MODE = 'NEW_ENVIRONMENT'
 
 export const NewAnalysisModal = Utils.withDisplayName('NewAnalysisModal')(
-  ({ isOpen, onDismiss, onSuccess, runtimes, apps, galaxyDataDisks, refreshRuntimes, refreshApps, workspace, persistentDisks, workspace: { workspace: { namespace, bucketName, name: workspaceName } } }) => {
+  ({ isOpen, onDismiss, onSuccess, uploadFiles, openUploader, runtimes, apps, galaxyDataDisks, refreshRuntimes, refreshApps, refreshAnalyses, analyses, workspace, persistentDisks, workspace: { workspace: { namespace, bucketName, name: workspaceName } } }) => {
     const [viewMode, setViewMode] = useState(undefined)
     const [busy, setBusy] = useState()
+    const [notebookKernel, setNotebookKernel] = useState('python3')
+    const [nameTouched, setNameTouched] = useState(false)
+    const [analysisName, setAnalysisName] = useState('')
+    const [currentTool, setCurrentTool] = useState(undefined)
 
-    const renderNewRuntimeModal = tool => h(NewRuntimeModalBase, {
-      isOpen: viewMode === NEW_JUPYTER_MODE || viewMode === NEW_RSTUDIO_MODE,
+
+    const currentRuntime = getCurrentRuntime(runtimes)
+    const currentRuntimeTool = currentRuntime?.labels?.tool
+    const currentApp = getCurrentApp(apps)
+
+    const resetView = () => {
+      setViewMode(undefined)
+      setAnalysisName('')
+      setCurrentTool(undefined)
+      setNotebookKernel('python3')
+    }
+
+    const enterNextViewMode = (currentTool, baseViewMode = viewMode) => {
```

Comment:
I'm concerned that the function argument `baseViewMode` having a default argument can make it easy to call that method without carefully thinking what value should be passed into `baseViewMode` (the same reason why we try to avoid default function argument values in IA codebases).

Reply:
> Hmm. My thoughts were the default (and currently predominant) use case is to call it with just the tool and no view mode, and have it be smart enough to figure out what should be next. Only in specific cases, where want to go out of the normal use case order due to an edge case and skip around in the modals is purposely passing a view mode applicable.

Fair enough :)

Reply:
> To be honest, this is the culmination of something @petesantos and I discussed a while ago. Tl;dr, a modal wizard that uses an object built of numbered steps representing modals to allow more free transitions. This seems like a good candidate, but I also feel it's not necessarily complex enough to warrant it yet.

I'd be curious to know more about this.